### ISSUE TITLE
DateTimeImmutable Doctrine issue

### ISSUE THREAD

**Issue by basz** [zf-fr/zfr-oauth2-server-doctrine#6]
Hi Michael,

tl;dr; Doctrine is not capable of working with DateTimeImmutable nor won't it be without dropping 5.4 support. Lets change DateTimeImmutable into DateTime.

This https://github.com/zf-fr/zfr-oauth2-server/blob/master/src/Model/AbstractToken.php#L51 conflicts with https://github.com/zf-fr/zfr-oauth2-server-doctrine/blob/master/config/doctrine/ZfrOAuth2.Server.Model.AbstractToken.dcm.xml#L15

Don't ask me why I only now see this issue - never seen Doctrine having trouble with DateTimeImmutable while building this module...

Adding custom types is a solution that works: For example (partially, needs a PR) https://github.com/VasekPurchart/Doctrine-Date-Time-Immutable-Types

But then either ALL DateTime will be handled as DateTimeImmutable which is likely to conflict with other (external) mapping definitions OR we must change the field definition from datetime to datetime_immutable and introduce a semi-hard dependancy on said solution.

Therefore - and the expiresAt property isn't exported uncloned from the model anyway - I propose to simply change to a DateTime.

Thoughts?

**Comment by basz** [comment]
I now know why this wasn't an issue before. doctrine/dbal dev-master is less forgiving when it comes handling these conversions it seems. It specifically check for instanceof DateTime which pre master did not. sigh...
